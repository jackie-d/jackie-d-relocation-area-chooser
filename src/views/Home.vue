<template>
  <mdb-container fluid style="height: 100%">
    <navbar />
    <div class="home">
        <mdb-card class="card-image" :style="{ backgroundImage: 'url(' + require('@/assets/gradient1.jpg') + ')' }">
          <div class="text-white text-center py-5 px-4 my-5">
            <div>
              <h2 class="card-title h1-responsive pt-3 mb-5 font-bold"><strong>{{ $t('dashboard.welcome') }}</strong></h2>
              <p class="mx-5 mb-5">
                {{ $t('dashboard.description') }}
              </p>
              <mdb-btn v-on:click="start" outline="white" size="md" icon="search-location">{{ $t('dashboard.start') }}</mdb-btn>
            </div>
          </div>
        </mdb-card>
    </div>
    <mainFooter />
  </mdb-container>
</template>

<script lang="ts">
import Vue from 'vue';

import { mdbContainer, mdbCard, mdbBtn } from 'mdbvue';
import navbar from '@/components/navbar.vue';
import mainFooter from '@/components/main-footer.vue';

import router from '../router/index';

export default Vue.extend({
  name: 'Home',
  components: {
    mdbContainer,
    mdbCard, 
    mdbBtn,
    navbar,
    mainFooter
  },
  methods: {
    start: function() {
      router.push('name');
    }
  }
});
</script>
