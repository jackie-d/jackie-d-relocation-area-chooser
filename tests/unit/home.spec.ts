import { shallowMount } from '@vue/test-utils'
import Home from '@/views/Home.vue';

describe('Home', () => {
  it('Renders title', () => {
    const wrapper = shallowMount(Home, {
      mocks: {
        $t: (key) => key
      }
    });
    expect(wrapper.find('h2').text()).toMatch('dashboard.welcome')
  })

  it('Renders subtitle', () => {
    const wrapper = shallowMount(Home, {
      mocks: {
        $t: (key) => key
      }
    });
    expect(wrapper.find('p').text()).toMatch('dashboard.description')
  })
  it('Snapshot consistence', () => {
    const wrapper = shallowMount(Home, {
        mocks: {
          $t: (key) => key
        }
      });
      expect(wrapper).toMatchSnapshot();
  });

})