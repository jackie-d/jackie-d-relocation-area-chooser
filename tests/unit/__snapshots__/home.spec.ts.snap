// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshot consistence 1`] = `
<div>
  <navbar-stub></navbar-stub>
  <mdb-container-stub tag="div" fluid="true" style="height: 100%;">
    <div class="home" style="margin: 0px -15px;">
      <mdb-card-stub tag="div" class="card-image" style="background-image: url(); background-size: cover; background-position: right bottom; min-height: 90vh;">
        <div class="text-white text-center py-5 px-4 my-5">
          <div>
            <h2 class="card-title h1-responsive pt-5 mb-5 font-weight-bold"><strong>dashboard.welcome</strong></h2>
            <p class="mx-5 mb-5 font-weight-bold">
              dashboard.description
            </p>
            <div>
              <mdb-btn-stub waves="true" tag="button" color="primary" size="lg" type="button" icon="play">dashboard.start</mdb-btn-stub>
            </div>
            <div class="mt-5 mb-5">
              <mdb-btn-stub waves="true" tag="button" color="primary" size="md" type="button" icon="caret-square-left">dashboard.history</mdb-btn-stub>
            </div>
          </div>
        </div>
      </mdb-card-stub>
    </div>
  </mdb-container-stub>
  <mainfooter-stub></mainfooter-stub>
</div>
`;
